### DISCUSSION TITLE
Define seed for training/finetuning processes and for a reproducible forecast

### DISCUSSION THREAD

**Discussion by dafmdev** [time-series-foundation-models/lag-llama#94] [Q&A]
How could a seed be defined for the reproducibility of results?

**Comment by ashok-arjun** [comment]
Sorry for the delay I didn't monitor Github discussions until now. 

I believe you've to add the random seed function before [this line](https://colab.research.google.com/drive/1XxrLW9VGPlZDw3efTvUi0hQimgJOwQG6?usp=sharing#scrollTo=RtVSLpmVSvzr).

I recommend seeding everything (`random`, `torch`, `numpy` etc.). I usually use this (but I haven't tested it for your usecase):

```
import os
import random
import numpy as np
import torch


def set_seed(seed, deterministic=True):
    random.seed(seed)
    np.random.seed(seed)
    torch.manual_seed(seed)
    if torch.cuda.is_available():
        torch.cuda.manual_seed(seed)
        torch.cuda.manual_seed_all(seed)
        torch.backends.cudnn.deterministic = deterministic
        torch.backends.cudnn.benchmark = False
    os.environ["PYTHONHASHSEED"] = str(seed)
```
